contracts/pco-license/facets/CFAReclaimerFacet.sol

```diff
@@ -1,28 +1,116 @@
+    event LicenseReclaimed(address indexed to, uint256 price);
+
+    /**
+     * @notice Current price to reclaim
+     */
+    function claimPrice() public view returns (uint256) {
+        require(
+            !LibCFABasePCO._isPayerBidActive(),
+            "CFAReclaimerFacet: Can only perform action when payer bid is active"
+        );
+        LibCFABasePCO.DiamondStorage storage ds = LibCFABasePCO
+            .diamondStorage();
+        LibCFABasePCO.DiamondCFAStorage storage cs = LibCFABasePCO.cfaStorage();
+
+        LibCFABasePCO.Bid storage _currentBid = LibCFABasePCO._currentBid();
+
+        uint256 originalForSalePrice = _currentBid.forSalePrice;
+        (uint256 startTime, , , ) = cs.cfaV1.cfa.getAccountFlowInfo(
+            ds.paramsStore.getPaymentToken(),
+            ds.paramsStore.getBeneficiary()
+        );
+        uint256 _length = ds.paramsStore.getReclaimAuctionLength();
+
+        if (block.timestamp > startTime + _length) {
+            return 0;
+        }
+
+        uint256 timeElapsed = block.timestamp - startTime;
+        uint256 priceDecrease = (originalForSalePrice * timeElapsed) / _length;
+        return originalForSalePrice - priceDecrease;
+    }
+
+    /**
+     * @notice Reclaim an inactive license as msg.sender
+     *      - Payer bid must be inactive
+     *      - Must have permissions to create flow for bidder
+     *      - Must have ERC-20 approval of payment token for claimPrice amount
+     * @param newContributionRate New contribution rate for license
+     * @param newForSalePrice Intented new for sale price. Must be within rounding bounds of newContributionRate
+     */
+    function claim(int96 newContributionRate, uint256 newForSalePrice)
+        external
+    {
+        require(
+            !LibCFABasePCO._isPayerBidActive(),
+            "CFAReclaimerFacet: Can only perform action when payer bid is active"
+        );
+
+        LibCFABasePCO.DiamondStorage storage ds = LibCFABasePCO
+            .diamondStorage();
+        LibCFABasePCO.DiamondCFAStorage storage cs = LibCFABasePCO.cfaStorage();
+
+        uint256 perSecondFeeNumerator = ds
+            .paramsStore
+            .getPerSecondFeeNumerator();
+        uint256 perSecondFeeDenominator = ds
+            .paramsStore
+            .getPerSecondFeeDenominator();
+
+        require(
+            LibCFABasePCO._checkForSalePrice(
+                newForSalePrice,
+                newContributionRate,
+                perSecondFeeNumerator,
+                perSecondFeeDenominator
+            ),
+            "CFAReclaimerFacet: Incorrect for sale price"
+        );
+
+        // Collect deposit
+        uint256 _claimPrice = claimPrice();
+        LibCFABasePCO.Bid storage _currentBid = LibCFABasePCO._currentBid();
+        address _bidder = _currentBid.bidder;
+        ISuperToken paymentToken = ds.paramsStore.getPaymentToken();
+        bool success = paymentToken.transferFrom(
```

Comment:
This should be updated to use `SafeERC20` from OpenZeppelin. I recently updated transfers in other facets to use this.

https://github.com/Geo-Web-Project/core-contracts/blob/main/contracts/pco-license/libraries/LibCFAPenaltyBid.sol#L166